pxr/base/arch/defines.h

```diff
@@ -104,4 +104,45 @@
 #define ARCH_HAS_MMAP_MAP_POPULATE
 #endif
 
+//
+// Sanitizers
```

Comment:
Can you explain a little more has these work in the comment? It's not immediately clear if these are checking if the sanitizer is enabled or just if the compiler version supports sanitizing.

Reply:
I added documentation to both `attributes.h` and `defines.h` to describe the role that both `ARCH_SANITIZE_ADDRESS` and `ARCH_NO_SANITIZE_ADDRESS_FUNCTION` play